src/main/scala/org/clulab/wm/eidos/attachments/NegationHandler.scala

```diff
@@ -178,7 +178,7 @@ class NegationHandler(val language: String) {
     for {
       (ix, lemma) <- leftContext ++ rightContext
       if !argumentIntervals.exists(_.contains(ix))
-      if (Seq("fail", "not") contains lemma) && !(previouslyFound contains ix)
+      if (Seq("fail", "not") contains lemma) && !(previouslyFound contains ix) && !(event.sentenceObj.words(ix+1) == "only")
```

Comment:
Can it be the case that `words(ix + 1)` is off the end?  It's language and weird things happen.  While this line is being touched, can that Seq be constructed once outside the loop in case it is being repeatedly generated?  It's old code, but better now than never.  It could be as far away as the `object NegationHandler` and be done just once ever.  For that matter, it may as well be an Array.
```scala
val failNot = Seq("fail", 'not")
...
if (failNot contains lemma)...
```
So both "not only" and "fail only" will be caught here.  Is that expected?

Reply:
I think this originally was intended to catch cases like "failed to X" and "did not X", but that did run into problems with "not only X", which we want to not put a negation attachment on. 

It is sort of hacky in the sense that I don't expect "fail not" to show up anywhere, but this does address the bug. I suppose we could get "fail only" in a context where we *do* want a negation attachment, but I can't think of an obvious case. Again, language *is* weird, so it might be good to separate those cases, which might mean separating `failNot` into two separate conditions. One that checks `if ("fail" == lemma)` and another one that checks `if ("not" == lemma && event.sentenceObj.words(ix+1) != "only")`.

Which would be better here, do you think?

Reply:
I'm inclined to assert linguist's prerogative on your behalf and let you decide.  The tighter check comes at a computational cost and if the consequence of getting it wrong is minimal, I can see skipping it, although perhaps it would be good to record the decision in a comment.

I would still be worried about the ix + 1 because that might crash the program.

The `import` would generally be skipped and instead the line written
```scala
if (NegationHandler.failNot contains lemma)
```
so that imports name nouns (classes and objects) rather than verbs (methods).  Picky, I know.

Reply:
Picky is good! I am but a humble scala amateur, so it's good to learn some best practices.

I think we can leave them combined for now, then if we get another complaint about a bug where "fail only" shows up we can adjust it then. I'll leave a comment about it though.

Agree about `ix + 1`. Without it being too convoluted, is there a good way to make sure that the `ix` is not the end of the sequence?

Something like the following to make sure the `ix + 1` is not beyond the last index of the sentence.
```
// if rightContext is empty OR if ix != last entry of right context (to avoid ix+1 error at end of sent)
if rightContext.isEmpty || (rightContext.nonEmpty && rightContext.last._1 < sentenceWords.lastIndexOf(sentenceWords.last))
if (NegationHandler.failNot contains lemma) && !(previouslyFound contains ix) && event.sentenceObj.words(ix+1) != "only"
```
If I understand my own code correctly, this will allow us to do `ix + 1` when the rightContext is empty (meaning we're only looking to the left of the trigger, so no worries about being off the end) or when the index at the end of the rightContext is less than the index of the end of the sentence (meaning we will still have at least one token to `ix + 1` into).

Can you see any glaring oversights with this?